test/framework/e2e_framework.go

```diff
@@ -253,6 +254,42 @@ func (f *E2eFramework) DeployK8sContextsSecret(namespace string) error {
 	return f.kustomizeAndApply(dir, namespace, f.ControlPlaneContext)
 }
 
+func (f *E2eFramework) DeployK8sClientConfigs(namespace string) error {
+	// Read from disk whatever the thing deployed..
+	kubeConfigFile := filepath.Join("..", "..", "build", "test-config", "k8s-contexts", "kubeconfig")
+
+	b, err := ioutil.ReadFile(kubeConfigFile)
+	if err != nil {
+		return err
+	}
+
+	apiConfig, err := clientcmd.Load(b)
+	if err != nil {
+		return err
+	}
+	for _, ctx := range apiConfig.Contexts {
+		// Deploy Secrets from here
+		cCfg := api.ClientConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      ctx.Cluster,
+				Namespace: namespace,
+			},
+			Spec: api.ClientConfigSpec{
+				ContextName: ctx.Cluster,
+				KubeConfigSecret: corev1.LocalObjectReference{
+					Name: "k8s-contexts",
```

Comment:
If we are creating a secret per kubeconfig, doesn't `Name` need to be distinct here and match the name of the secrets? I think you need to update the `generateContextsKustomization` function. 

Note that one of the intermediate steps of `setup-kind-multicluster.sh` is to write out the individual kubeconfig files. You want the updated ones intended for in-cluster use which are located at `build/kubeconfigs/updated`.

Reply:
In this case, the same Secret is used for all the ClientConfigs, that's why the name was hardcoded for e2e-tests.